packetbeat/config/agent.go

```diff
@@ -0,0 +1,80 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package config
+
+import (
+	"fmt"
+	"runtime"
+	"strings"
+
+	"github.com/elastic/beats/v7/libbeat/common"
+	"github.com/elastic/beats/v7/libbeat/logp"
+)
+
+var osDefaultDevices = map[string]string{
+	"darwin": "en0",
+	"linux":  "any",
+}
+
+func defaultDevice() string {
+	if device, found := osDefaultDevices[runtime.GOOS]; found {
+		return device
+	}
+	return "0"
+}
+
+// Normalize allows the packetbeat configuration to understand
+// agent semantics
+func (c Config) Normalize() (Config, error) {
+	logp.Debug("agent", "Normalizing agent configuration")
+	if len(c.Inputs) > 0 {
+		// override everything, we're managed by agent
+		c.Flows = nil
+		c.Protocols = nil
+		c.ProtocolsList = []*common.Config{}
+		// TODO: make this configurable rather than just using the default device in
```

Comment:
That seems important 😄 .

Reply:
Yep, definitely important. Since this code is only hit when agent-based config passes in configuration under the (undocumented) `inputs` array, I figured we could just do something sane-ish for now. I called it out in the PR description, but we'll want to discuss how we want to configure interfaces going forward since historically packetbeat creates a single sniffer on a given interface and then passes packets on to all the registered protocol dissectors after that. We'll need to decide how we can either change that or if we need to make a single protocol sniffing package that has this as a global conffiguration option sometime in the future.